qt-bugs@ issue : none
Trolltech task ID : None
applied: no
author: Alexis Menard <alexis.menard@trolltech.com>

Fix deletion of a qgraphicswidget on clear focus even if it doesn't have the focus.

Will be included in 4.4.4

Index: ../src/gui/graphicsview/qgraphicsitem.cpp
===================================================================
--- src/gui/graphicsview/qgraphicsitem.cpp   (revision 883124)
+++ src/gui/graphicsview/qgraphicsitem.cpp   (working copy)
@@ -1951,7 +1951,7 @@
 */
 void QGraphicsItem::clearFocus()
 {
-    if (!d_ptr->scene || !hasFocus())
+    if (!d_ptr->scene)
         return;
     if (d_ptr->isWidget) {
         // Invisible widget items with focus must explicitly clear subfocus.

